area/guadosalam.py

```diff
@@ -5,123 +5,71 @@
-            pathing.set_movement([18, -119])
-            xbox.tap_b()
-        FFXC.set_neutral()
-        memory.main.click_to_control_3()
-
-    else:
-        while not pathing.set_movement([4, -114]):
-            pass
-        logger.debug("Talk to Auron")
-        while memory.main.user_control():  # Talk to Auron (first for affection)
-            pathing.set_movement([18, -119])
-            xbox.tap_b()
-        FFXC.set_neutral()
-        memory.main.click_to_control_3()
-
-        while not pathing.set_movement([-39, -77]):
-            pass
-        logger.debug("Wakka conversation")
-        while memory.main.user_control():  # Start conversation with Wakka
-            pathing.set_movement([-49, -61])
-            xbox.tap_b()
-        FFXC.set_neutral()
-        memory.main.click_to_control_3()
-
-        while not pathing.set_movement([-13, -67]):
-            pass
-        logger.debug("Lulu conversation")
-        while memory.main.user_control():  # Lulu conversation
-            pathing.set_movement([-11, -55])
+    while memory.main.get_map() != 141: #Up to the dining hall scenes
+        if memory.main.user_control():
+            if checkpoint == 4:
+                # Into the first door
+                FFXC.set_movement(0,1)
+                memory.main.await_event()
+                checkpoint += 1
+            elif checkpoint == 6:
+                # Back out the door
+                FFXC.set_movement(0,-1)
+                memory.main.await_event()
+                checkpoint += 1
+            elif checkpoint == 8:
+                # Into the dining hall
+                FFXC.set_movement(0,1)
+                memory.main.await_event()
+                checkpoint += 1
+            elif pathing.set_movement(GuadoStart.execute(checkpoint)):
+                checkpoint += 1
+                logger.debug(f"Checkpoint {checkpoint}")
+        else:
+            FFXC.set_neutral()
             xbox.tap_b()
-        FFXC.set_neutral()
-        memory.main.click_to_control_3()
-
-    # Line up for Rikku/Yuna
-    while not pathing.set_movement([15, -52]):
-        pass
-
-    while not pathing.set_movement([22, -25]):
-        pass
-    logger.debug("Rikku conversation")
-    while memory.main.user_control():  # Start conversation with Rikku
-        pathing.set_movement([8, -26])
-        xbox.tap_b()
-    FFXC.set_neutral()
-    memory.main.click_to_control_3()
-
-    while not pathing.set_movement([27, -37]):
-        pass
-    logger.debug("Yuna conversation")
-    while memory.main.user_control():  # Yunas turn
-        pathing.set_movement([39, -33])
-        xbox.tap_b()
-    FFXC.set_neutral()
-    memory.main.click_to_control_3()
+    
+    # Checkpoint carries over.
+    logger.info("Now in the room with all the food")
+    while not memory.main.get_map() == 163:
+        if memory.main.user_control():
+            if checkpoint == 10:
+                pathing.approach_actor(actor_id=3)
```

Comment:
Are the `actor_id`s for players always the same? Is Auron always `actior_id=whatever` no matter what map he is on? If so, we can add the `actor_id` to the player class and then do `pathing.approach_actor(Auron)` and have `approach_actor detect it's a player class and pull the actor id off the player class.

It looks like in discord the `actor_id` is always one more than the character id. If that's the case, then we can do something like:


```class Player:
    ...
    @property
    def actor_id(self):
        return self.id + 1
```

Then we can do something like `Auron.actor_id` and it'll return the correct actor id.

I'll be pitching more changes related to `approach_actor` further down. So I'd avoid making changes until the code review is done.

Reply:
Agree with your conclusion. There is a lot we can do with actors (ref Peppy's Farenheit project) when we land the Nemesis route. I am more or less holding back for that reason only.

https://github.com/fkelava/fahrenheit/blob/master/Fahrenheit.Core.X/Kernel/FhXChrPc.cs